content/hardware/08.edu/solution-and-kits/alvik/tutorials/user-manual/user-manual.md

```diff
@@ -240,6 +346,1018 @@ Open **Arduino Lab for MicroPython** and **connect** Alvik. Then:
+`angular_velocity`: Speed of the wheels.
+`linear_unit`: Unit of linear velocity (default is 'cm/s').
+`angular_unit`: Unit of rotational speed of the wheels (default is 'deg/s').
+
+**Example Usage:**
+
+```python
+alvik.drive(10, 30, 'cm/s', 'deg/s')
+sleep_ms(2000)  # Drive at 10 cm/s linear velocity and 30 deg/s angular velocity for 2 seconds
+alvik.brake()
+```
+
+### Encoder’s Control
+
+The Alvik robot provides various functions to control and monitor the motors using encoders. These functions allow you to get real-time feedback on the speed, position, and pose of the robot, which is essential for precise movement and navigation.
+
+** Check Wheels Speed **
+
+The `get_wheels_speed` function returns the current speed of the wheels.
+
+**Inputs:**
+- `unit`: Unit of rotational speed of the wheels (default is 'rpm').
+
+**Outputs:**
+- `left_wheel_speed`: The speed of the left wheel.
+- `right_wheel_speed`: The speed of the right wheel.
+
+**Example Usage:**
+```python
+left_speed, right_speed = alvik.get_wheels_speed(unit='rpm')
+print(f"Left Wheel Speed: {left_speed} rpm, Right Wheel Speed: {right_speed} rpm")
+```
+
+**Check Wheels Position**
+
+The `get_wheels_position` function returns the current angle of the wheels.
+
+**Inputs:**
+- `unit`: Unit of the wheel angle (default is 'deg').
+
+**Outputs:**
+- `left_wheel_angle`: The angle of the left wheel.
+- `right_wheel_angle`: The angle of the right wheel.
+
+**Example Usage:**
+```python
+left_angle, right_angle = alvik.get_wheels_position(unit='deg')
+print(f"Left Wheel Angle: {left_angle} degrees, Right Wheel Angle: {right_angle} degrees")
+```
+
+**Check Velocity**
+
+The `get_drive_speed` function returns the linear and angular velocity of the robot.
+
+**Inputs:**
+- `linear_unit`: Unit of linear velocity (default is 'cm/s').
+- `angular_unit`: Unit of rotational speed of the wheels (default is 'deg/s').
+
+**Outputs:**
+- `linear_velocity`: The linear speed of the robot.
+- `angular_velocity`: The angular speed of the wheels.
+
+**Example Usage:**
+```python
+linear_velocity, angular_velocity = alvik.get_drive_speed(linear_unit='cm/s', angular_unit='deg/s')
+print(f"Linear Velocity: {linear_velocity} cm/s, Angular Velocity: {angular_velocity} deg/s")
+```
+
+**Check Pose**
+
+The `get_pose` function returns the current pose of the robot.
+
+**Inputs:**
+- `distance_unit`: Unit of x and y outputs (default is 'cm').
+- `angle_unit`: Unit of theta output (default is 'deg').
+
+**Outputs:**
+- `x`: The x-coordinate of the robot.
+- `y`: The y-coordinate of the robot.
+- `theta`: The orientation angle of the robot.
+
+**Example Usage:**
+```python
+x, y, theta = alvik.get_pose(distance_unit='cm', angle_unit='deg')
+print(f"Pose - X: {x} cm, Y: {y} cm, Theta: {theta} degrees")
+```
+
+These functions allow for precise control and monitoring of the Alvik robot's movements, enabling advanced navigation and path planning capabilities.
+
+
+### Reading Buttons

```

Comment:
the links in this section are not working

Reply:
The links for the buttons are working when not in preview